### PR TITLE
ExpenseList CRUD (without Edit) and improve Auth

### PR DESCRIPTION
**Pull request by galdd**

### CODE REVIEW

File: src/db/@types.ts [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
 export type ID = {
-    id: string;
-}
\ No newline at end of file
+  id: string;
+};
+
+export interface UserRequest extends Request {
+  user?: {
+    sub: string;
```
Comment (ofir-zeitoun) [inline src/db/@types.ts:10]: what does sub refer to?

File: src/features/expenses-list/expenses-list.routes.ts [hunk 2]
```diff
@@ -1,59 +1,109 @@
...
 import { ExpensesListModel } from "./expenses-list.model";
 
 export const router = Router();
 
 router.get(
   "/",
   validateResource(paginationSchema),
-  async (req: Request<any, any, any, Pagination>, res: Response) => {
-    console.log("Received request for fetching expense lists.");
+  async (req: UserRequest, res: Response, next: NextFunction) => {
```
Comment (ofir-zeitoun) [inline src/features/expenses-list/expenses-list.routes.ts:19]: do we need next?

File: src/features/expenses-list/expenses-list.routes.ts [hunk 3]
```diff
@@ -1,59 +1,109 @@
...
+    const offsetNumber = parseInt(req.query.offset as string);
+    const limitNumber = parseInt(req.query.limit as string);
 
-    const offsetNumber = parseInt(req.query.offset);
-    const limitNumber = parseInt(req.query.limit);
+    const lists = await ExpensesListModel.find({})
+      .populate("creator", "name image")
```
Comment (ofir-zeitoun) [inline src/features/expenses-list/expenses-list.routes.ts:24]: do we need both ?

File: src/features/expenses-list/expenses-list.routes.ts [hunk 4]
```diff
@@ -1,59 +1,109 @@
...
+
+      if (!deletedList) {
+        return res.status(status.NOT_FOUND).json({ message: "List not found" });
+      }
+
+      res.status(status.OK).json({ message: "List deleted successfully" });
     } catch (error) {
-      console.error("Failed to fetch expenses lists:", error);
-      res.status(status.INTERNAL_SERVER_ERROR).send("Internal Server Error");
+      next(error);
```
Comment (ofir-zeitoun) [inline src/features/expenses-list/expenses-list.routes.ts:106]: why next?
remove try catch

File: src/features/users/users.routes.ts [hunk 5]
```diff
@@ -54,5 +54,40 @@ router.delete(
     res.status(status.OK).json(deleteUser);
   }
 );
+router.post("/check", async (req: Request, res: Response) => {
+  const { auth0Id, name, email, picture } = req.body;
```
Comment (ofir-zeitoun) [inline src/features/users/users.routes.ts:58]: can we extract autoId from token?

File: src/routes/middlewares/error.ts [hunk 6]
```diff
@@ -1,7 +1,16 @@
...
 import status from "http-status";
 
-export const errorHandler = (err: TypeError, _req: Request, res: Response) => {
+export const errorHandler = (
+  err: TypeError,
+  _req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (res.headersSent) {
```
Comment (ofir-zeitoun) [inline src/routes/middlewares/error.ts:10]: what is this change for?

File: src/routes/middlewares/validate-resource.ts [hunk 7]
```diff
@@ -2,14 +2,19 @@ import { NextFunction, Request, Response } from "express";
 import status from "http-status";
 import { AnyZodObject } from "zod";
 
+interface CustomRequest extends Request {
```
Comment (ofir-zeitoun) [inline src/routes/middlewares/validate-resource.ts:5]: interesting...

File: src/db/@types.ts (hunk 1 above)
Comment (galdd) [reply src/db/@types.ts:10]: Refers to the "Subject Identifier", which is a unique identifier for each user
 * obtained from the JWT issued by Auth0

File: src/features/expenses-list/expenses-list.routes.ts (hunk 2 above)
Comment (galdd) [reply src/features/expenses-list/expenses-list.routes.ts:19]: not neccesery, removed

File: src/features/expenses-list/expenses-list.routes.ts (hunk 3 above)
Comment (galdd) [reply src/features/expenses-list/expenses-list.routes.ts:24]: not necessary, removed

File: src/features/expenses-list/expenses-list.routes.ts (hunk 4 above)
Comment (galdd) [reply src/features/expenses-list/expenses-list.routes.ts:106]: not necessary, removed

File: src/features/users/users.routes.ts (hunk 5 above)
Comment (galdd) [reply src/features/users/users.routes.ts:58]: yes, changed

File: src/features/expenses-list/expenses-list.routes.ts (hunk 3 above)
Comment (galdd) [reply src/features/expenses-list/expenses-list.routes.ts:24]: no,removed

File: src/routes/middlewares/error.ts (hunk 6 above)
Comment (galdd) [reply src/routes/middlewares/error.ts:10]: I had some issiue that i tried to solve, but not necesery anymore, return to original

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**